### ISSUE TITLE
Error: undeclared identifier: 'log'

### ISSUE THREAD

**Issue by syhpoon** [status-im/nim-chronicles#52]
Hey guys,

I'm trying to incorporate chronicles into my library and found a weird compilation-time error.
After stripping all the nonrelevant parts, this is what I've got:

file a.nim:
```nim
import                                                                                             
   asyncdispatch,                                                                                  
   chronicles                                                                                      
                                                                                                   
type                                                                                               
   Wut*[P] = ref object                                                                            
      ps: seq[P]                                                                                   
                                                                                                   
proc newWut*[P](ps: seq[P]): Wut[P] =                                                              
   Wut[P](ps: ps)                                                                                  
                                                                                                   
proc action(d: Wut) {.async.} =                                                                    
   error "ERROR"                                                                                   
   #echo("ERROR")                                                                                  
                                                                                                   
proc runForever*(d: Wut) {.async.} =                                                               
   while true:                                                                                     
      asyncCheck d.action()
```

and file main.nim:
```nim
import asyncdispatch                                                                               
import a                                                                                           
                                                                                                   
let d = newWut[int](@[1])                                                                          
                                                                                                   
asyncCheck d.runForever()                                                                          
runForever()
```

when running `nim c main.nim`, I'm getting the following error:
```
main.nim(6, 13) template/generic instantiation from here
a.nim(18, 19) template/generic instantiation from here
a.nim(13, 10) template/generic instantiation from here
/home/user/.nimble/pkgs/chronicles-0.5.1/chronicles.nim(344, 5) Error: undeclared identifier: 'log'
```

If I replace    `error "ERROR"` with `echo("ERROR")`, everything compiles successfully.
My nim version is:
```Nim Compiler Version 0.19.4 [Linux: amd64]
Compiled at 2019-02-01
Copyright (c) 2006-2018 by Andreas Rumpf

git hash: b6d96cafc8bcad1f3d32f2910b25cd11a93f7751
active boot switches: -d:release```

**Comment by zah** [comment]
This is a known Nim issue related to the fact that the functions `proc runForever*(d: Wut)` and `proc action(d: Wut)` are generic. Nim has some issues with generic symbol look-ups inside modules that are "sandwiched" between the outermost instantiation scope (main.nim) and the innermost non-generic resolved call (chronicles.nim). I'm sorry if this explanation is a bit vague and hard to follow, but the short version of the story is that you can fix the problem by importing chronicles in main.nim:

```nim
import asyncdispatch, chronicles 
import a  

let d = newWut[int](@[1])   

asyncCheck d.runForever()
runForever()
```

**Comment by syhpoon** [comment]
Thank you, @zah ! Now it works.

**Comment by zah** [comment]
The underlying Nim issue will be tracked here:
https://github.com/nim-lang/Nim/issues/11225
